src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -858,61 +859,55 @@ ss::future<checked<tm_transaction, tx_errc>> tx_gateway_frontend::reabort_tm_tx(
 }
 
 // get_tx must be called under stm->get_tx_lock
-ss::future<checked<tm_transaction, tx_errc>> tx_gateway_frontend::get_tx(
+ss::future<checked<tm_transaction, tx_errc>>
+tx_gateway_frontend::get_ongoing_tx(
   ss::shared_ptr<tm_stm> stm,
   model::producer_identity pid,
   kafka::transactional_id transactional_id,
   model::timeout_clock::duration timeout) {
     auto maybe_tx = stm->get_tx(transactional_id);
     if (!maybe_tx) {
-        return ss::make_ready_future<checked<tm_transaction, tx_errc>>(
-          tx_errc::timeout);
+        co_return checked<cluster::tm_transaction, tx_errc>(tx_errc::timeout);
     }
 
     auto tx = maybe_tx.value();
 
     if (tx.pid != pid) {
-        return ss::make_ready_future<checked<tm_transaction, tx_errc>>(
-          tx_errc::timeout);
-    }
-
-    if (tx.status == tm_transaction::tx_status::aborting) {
-        return ss::make_ready_future<checked<tm_transaction, tx_errc>>(
-          tx_errc::timeout);
+        co_return checked<cluster::tm_transaction, tx_errc>(tx_errc::timeout);
     }
 
-    auto f = ss::make_ready_future<checked<tm_transaction, tx_errc>>(tx);
+    checked<tm_transaction, tx_errc> r(tx);
 
     if (tx.status != tm_transaction::tx_status::ongoing) {
         if (tx.status == tm_transaction::tx_status::preparing) {
-            f = do_commit_tm_tx(stm, tx, timeout, ss::promise<tx_errc>());
+            r = co_await do_commit_tm_tx(
+              stm, tx, timeout, ss::promise<tx_errc>());
         } else if (tx.status == tm_transaction::tx_status::prepared) {
-            f = recommit_tm_tx(tx, timeout);
+            r = co_await recommit_tm_tx(tx, timeout);
+        } else if (tx.status == tm_transaction::tx_status::aborting) {
+            r = co_await reabort_tm_tx(tx, timeout);
```

Comment:
this commit subject is `rename get_tx to get_ongoing_tx` but this patch seems to be doing quite a lot more than renaming a method: switching to co-routines, but also it looks like some logic is changing too.

Reply:
you're right, I've splited that commit